quickwit-core/src/index.rs

```diff
@@ -83,21 +84,23 @@ impl IndexService {
         &self,
         index_config: IndexConfig,
     ) -> Result<IndexMetadata, IndexServiceError> {
-        let index_uri = if let Some(index_uri) = index_config.index_uri.as_ref() {
-            index_uri.to_string()
+        let index_id = index_config.index_id.clone();
+        let index_uri = if let Some(index_uri) = &index_config.index_uri {
+            index_uri.clone()
         } else {
-            let default_index_uri =
-                format!("{}/{}", self.default_index_root_uri, index_config.index_id);
+            let index_uri = self.default_index_root_uri.join(&index_id).expect(
+                "Failed to create default index URI. This should never happen. Please, report.",
```

Comment:
Don't know if it's a good idea to put the github issue link.

Reply:
Might as well.